pkg/client/proxy.go

```diff
@@ -0,0 +1,91 @@
+package client
+
+import (
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+	"github.com/longhorn/longhorn-instance-manager/pkg/util"
+
+	emeta "github.com/longhorn/longhorn-engine/pkg/meta"
+)
+
+var (
+	ErrParameter = errors.Errorf("missing required parameter")
+)
+
+type ServiceContext struct {
+	cc *grpc.ClientConn
+
+	ctx  context.Context
+	quit context.CancelFunc
+
+	service rpc.ProxyServiceClient
+}
+
+func (c *ProxyClient) Close() error {
+	c.quit()
+	if err := c.cc.Close(); err != nil {
+		return errors.Wrapf(err, "error closing proxy gRPC connection")
+	}
+	return nil
+}
+
+type ProxyClient struct {
+	ServiceURL string
+	ServiceContext
+}
+
+func NewProxyClient(ctx context.Context, ctxCancel context.CancelFunc, address string, port int) (*ProxyClient, error) {
+	getServiceCtx := func(serviceUrl string) (ServiceContext, error) {
+		connection, err := grpc.Dial(serviceUrl, grpc.WithInsecure())
```

Comment:
We should adopt TLS here, because manager will communicate with instance manager.